src/shims/posix/linux/sync.rs

```diff
@@ -33,10 +31,11 @@ pub fn futex<'tcx>(
             if !this.is_null(timeout)? {
                 throw_ub_format!("miri does not support timeouts for futex operations");
             }
+            this.memory.check_ptr_access(addr_scalar, Size::from_bytes(4), Align::from_bytes(4).unwrap())?;
```

Comment:
Seems like you found out that always doing the check doesn't work... in such cases please always add a comment explaining what happens when and why. If you thought that was the most natural implementation, other readers will think the same, and will wonder why the code looks the way it does.